source/Core/SMV Code Generation/BasicFbSmv.cs

```diff
@@ -80,6 +89,26 @@ public static string EcStateChangeBlock(IEnumerable<ECTransition> transitions, I
                 }
                 return String.Format("\n" + Smv.NextCaseBlock + "\n", Smv.EccStateVar, ecTransitionsSmv);
             }
+
+            public static string NdtChangeBlock(IEnumerable<ECTransition> transitions)
+            {
+                string ecTransitionsSmv = "";
+
+                foreach (var transition in transitions)
+                {
+                    if (transition.Condition != null && transition.Condition != "1" && transition.Condition.Contains("NDT"))
+                    {
+                        ecTransitionsSmv += "\t";
+                        ecTransitionsSmv += Smv.EccStateVar + "=" + Smv.EccState(transition.Source);
+                        ecTransitionsSmv += " & ";
+                        ecTransitionsSmv += Smv.OsmStateVar + "=" + Smv.Osm.S1;
+                        ecTransitionsSmv += " : ";
+                        ecTransitionsSmv += "{TRUE,FALSE}";
+                        ecTransitionsSmv += ";\n";
+                    }
+                }
+                return String.Format("\n" + Smv.NextCaseBlock + "\n", "NDT", ecTransitionsSmv);
+            }
```

Comment:
NDT change block works ok, but may be in future this will be simplified like this.

**next(NDT) := { TRUE,FALSE}**

But need to test and verify the results after that.